mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,801 @@
+            Default: 0.
+        overlap_self (float): Percent of the proposal's own span contained
+            in a groundtruth instance. Default: 0.
+    """
+
+    def __init__(self,
+                 start_frame,
+                 end_frame,
+                 num_video_frames,
+                 label=None,
+                 best_iou=0,
+                 overlap_self=0):
+        self.start_frame = start_frame
+        self.end_frame = min(end_frame, num_video_frames + 1)
+        self.num_video_frames = num_video_frames
+        self.label = label if label is not None else -1
+        self.coverage = (end_frame - start_frame) / num_video_frames
+        self.best_iou = best_iou
+        self.overlap_self = overlap_self
+        self.loc_reg = None
+        self.size_reg = None
+        self.regression_targets = [0., 0.]
+
+    def compute_regression_targets(self, gt_list, positive_threshold):
+        """Compute regression targets of positive proposals.
+
+        Args:
+            gt_list (list): The list of groundtruth instances.
+            positive_threshold (float): Minimum threshold of overlap of
+                positive/foreground proposals and groundtruths.
+        """
+        # Find the groundtruth instance with the highest IOU.
+        ious = [
+            temporal_iou(self.start_frame, self.end_frame, gt.start_frame,
+                         gt.end_frame) for gt in gt_list
+        ]
+        best_gt = gt_list[np.argmax(ious)]
+
+        # interval: [start_frame, end_frame)
+        proposal_center = (self.start_frame + self.end_frame - 1) / 2
+        gt_center = (best_gt.start_frame + best_gt.end_frame - 1) / 2
+        proposal_size = self.end_frame - self.start_frame
+        gt_size = best_gt.end_frame - best_gt.start_frame
+
+        # Get regression targets:
+        # (1). Localization regression target:
+        #     center shift proportional to the proposal duration
+        # (2). Duration/Size regression target:
+        #     logarithm of the groundtruth duration over proposal duration
+
+        self.loc_reg = (gt_center - proposal_center) / proposal_size
+        self.size_reg = math.log(gt_size / proposal_size)
+        self.regression_targets = ([self.loc_reg, self.size_reg]
+                                   if self.loc_reg is not None else [0., 0.])
+
+
+@DATASETS.register_module()
+class SSNDataset(BaseDataset):
+    """Proposal frame dataset for Structured Segment Networks. Based on
+    proposal information, the dataset loads raw frames and apply specified
+    transforms to return a dict containing the frame tensors and other
+    information. The ann_file is a text file with multiple lines and each
+    video's information takes up several lines. This file can be a normalized
+    file with percent or standard file with specific frame indexes. If the file
+    is a normalized file, it will be converted into a standard file first.
+    Template information of a video in a standard file:
```

Comment:
```
Proposal frame dataset for Structured Segment Networks.

Based on proposal information, the dataset loads raw frames and applies specified
transforms to return a dict containing the frame tensors and other information.

The ann_file is a text file with multiple lines and each video's information takes up several lines.
The ann_file can be a normalized file with ‘normalized_’ as a prefix in the filename and has percent frame indexes, 
or a standard file with specific frame indexes. If the file is a normalized file, it will be converted into a standard file first.

Template information of a video in a standard file:
```

Reply:
Need to double check (especially the description of the difference between normalized file and standard file)